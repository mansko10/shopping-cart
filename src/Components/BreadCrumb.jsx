import PropTypes from "prop-types";
import StyledBreadCrumb from "./Styles/StyledBreadCrumb";
import createBreadCrumb from "../Util/createBreadCrumb";
import { Fragment } from "react";
import { Link } from "react-router-dom";

export default function BreadCrumb({ pathname }) {
  const breadcrumbs = createBreadCrumb(pathname);
  return (
    <StyledBreadCrumb>
      {breadcrumbs.map((breadcrumb, index) => {
        if (index !== breadcrumbs.length - 1) {
          return (
            <Fragment key={index}>
              <Link to={breadcrumb.path}>
                {decodeURIComponent(breadcrumb.name)}
              </Link>
              <span>{">"}</span>
            </Fragment>
          );
        } else {
          return (
            <Fragment key={index}>
              <Link to="" className="active">
                {decodeURIComponent(breadcrumb.name)}
              </Link>
            </Fragment>
          );
        }
      })}
    </StyledBreadCrumb>
  );
}

BreadCrumb.propTypes = {
  pathname: PropTypes.string,
};
