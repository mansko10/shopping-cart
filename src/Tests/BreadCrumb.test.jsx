import { it, expect, describe } from "vitest";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import BreadCrumb from "../Components/BreadCrumb";

describe("Renders BreadCrumb.jsx correctly", () => {
  it("Renders shop breadcrumb correctly", () => {
    const { container } = render(<BreadCrumb pathname="/shop" />, {
      wrapper: BrowserRouter,
    });

    expect(container).toMatchSnapshot();
  });

  it("Renders 'shop/mens' clothing' correctly", () => {
    const { container } = render(
      <BreadCrumb pathname="/shop/men's%20clothing" />,
      {
        wrapper: BrowserRouter,
      },
    );

    expect(container).toMatchSnapshot();
  });

  it("Renders 'shops/men's clothing/1 correctly", () => {
    const { container } = render(
      <BreadCrumb pathname="/shop/men's%20clothing/1" />,
      {
        wrapper: BrowserRouter,
      },
    );

    expect(container).toMatchSnapshot();
  });
});
