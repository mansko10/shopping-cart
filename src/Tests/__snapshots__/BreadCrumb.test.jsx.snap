// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders BreadCrumb.jsx correctly > Renders 'shop/mens' clothing' correctly 1`] = `
<div>
  <div
    class="sc-blHHSb eSnmzD"
  >
    <a
      data-discover="true"
      href="/shop/"
    >
      shop
    </a>
    <span>
      &gt;
    </span>
    <a
      data-discover="true"
      href="/shop/men's%20clothing/"
    >
      men's clothing
    </a>
    <span>
      &gt;
    </span>
  </div>
</div>
`;

exports[`Renders BreadCrumb.jsx correctly > Renders 'shops/men's clothing/1 correctly 1`] = `
<div>
  <div
    class="sc-blHHSb eSnmzD"
  >
    <a
      data-discover="true"
      href="/shop/"
    >
      shop
    </a>
    <span>
      &gt;
    </span>
    <a
      data-discover="true"
      href="/shop/men's%20clothing/"
    >
      men's clothing
    </a>
    <span>
      &gt;
    </span>
    <a
      data-discover="true"
      href="/shop/men's%20clothing/1/"
    >
      1
    </a>
    <span>
      &gt;
    </span>
  </div>
</div>
`;

exports[`Renders BreadCrumb.jsx correctly > Renders shop breadcrumb correctly 1`] = `
<div>
  <div
    class="sc-blHHSb eSnmzD"
  >
    <a
      data-discover="true"
      href="/shop/"
    >
      shop
    </a>
    <span>
      &gt;
    </span>
  </div>
</div>
`;
